source/server/listener_manager_impl.cc

```diff
@@ -554,8 +554,17 @@ void ListenerManagerImpl::startWorkers(GuardDog& guard_dog) {
 }
 
 void ListenerManagerImpl::stopListeners() {
+  auto counter = std::make_shared<std::atomic<uint32_t>>(workers_.size());
```

Comment:
nit: more meaningful name like `workers_pending_stopping` or some thing like that?